vendor/knative.dev/pkg/test/upgrade/shell/executor.go

```diff
@@ -186,3 +197,14 @@ func quoteArgs(args []string) string {
 	}
 	return strings.Join(quoted, " ")
 }
+
+func (w testingWriter) Write(p []byte) (n int, err error) {
+	n = len(p)
+
+	// Strip trailing newline because t.Log always adds one.
+	p = bytes.TrimRight(p, "\n")
+
+	w.t.Logf("%s", p)
```

Comment:
It would be a good idea to Log every line of the output. Just for ascetic reasons. Now some lines are skewed in the output:

```
executor.go:207: Oct 12 07:08:25.738 install_head_reuse_ingress [OUT] ***************************************
    Oct 12 07:08:25.738 install_head_reuse_ingress [OUT] ***         E2E TEST FAILED         ***
    Oct 12 07:08:25.738 install_head_reuse_ingress [OUT] ***    Start of information dump    ***
    Oct 12 07:08:25.738 install_head_reuse_ingress [OUT] ***************************************
executor.go:207: Oct 12 07:08:25.739 install_head_reuse_ingress [OUT] >>> The dump is located at /logs/artifacts/k8s.dump-.txt
executor.go:207: Oct 12 07:08:26.048 install_head_reuse_ingress [OUT] >>> allowlistedv2workloads.auto.gke.io (0 objects)
```
See: https://prow.knative.dev/view/gs/knative-prow/pr-logs/pull/knative_serving/14495/upgrade-tests_serving_main/1712360626715627520

Something like:

```suggestion
	for line := range strings.Split(p, "\n") {
		w.t.Log(line)
	}
```

Reply:
It's a cosmetic change but why not. I've tested this in https://prow.knative.dev/view/gs/knative-prow/pr-logs/pull/knative_serving/14495/upgrade-tests_serving_main/1713811249294217216.
Anyway, this change would be in knative.dev/pkg so I can do it later. It doesn't affect this PR.

Reply:
I've upated this PR and it's ready for review/merging.

Reply:
Done in https://github.com/knative/pkg/pull/2867